pkg/worker/job_utilities.go

```diff
@@ -4,12 +4,77 @@ import (
 	"context"
 
 	faktory_worker "github.com/contribsys/faktory_worker_go"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 
 	"github.com/cmsgov/mint-app/pkg/apperrors"
 )
 
-const faktoryLoggingSection string = "faktory"
-const appSectionKey string = "app_section"
+// faktoryAppSectionField provides the zap field for specifying the part of the application is faktory
+var faktoryAppSectionField = zap.String(appSectionKey, faktoryLoggingSection)
+
+// these constants represents the keys to get these data fields out of a zap logger.
+const (
+	faktoryLoggingSection string = "faktory"
+	appSectionKey         string = "app_section"
+
+	batchIDKey string = "BID"
+	jobIDKey   string = "JID"
+	jobTypeKey string = "job_type"
+
+	translatedAuditQueueIDKey = "translated_audit_queue_id"
+	auditQueueAttemptsKey     = "audit_queue_attempts"
+	auditChangeKey            = "change_id"
+	modelPlanIDKey            = "model_plan_id"
+	dateKey                   = "date"
+	userIDKey                 = "userID"
+)
+
+// BIDZapField returns the zap core field for a worker BatchID
+func BIDZapField(bid string) zapcore.Field {
+	return zap.String(batchIDKey, bid)
+}
+
+// JIDZapField returns the zap core field for a worker JobID
+func JIDZapField(jid string) zapcore.Field {
+	return zap.String(jobIDKey, jid)
+}
+
+// JobTypeZapField returns the zap core field for a worker job type
+func JobTypeZapField(jobType string) zapcore.Field {
+	return zap.String(jobTypeKey, jobType)
+}
+
+// TranslatedAuditQueueIDZapField provides the zap field for a TranslatedAuditQueueID
+func TranslatedAuditQueueIDZapField(translatedAuditQueueID uuid.UUID) zapcore.Field {
+	return zap.Any(translatedAuditQueueIDKey, translatedAuditQueueID)
+}
+
+// auditChangeIDZapField provides the zap field for an audit change id
+func auditChangeIDZapField(changeID interface{}) zapcore.Field {
+	return zap.Any(auditChangeKey, changeID)
+}
```

Comment:
I like the idea behind these functions (presumably to allow for consistent search/query in Splunk), but it does seem to result in some really long / difficult to read logger statements (when I'd prefer to try and keep them simple)

```go
logger.With(TranslatedAuditQueueIDZapField(queueObj.ID), auditChangeIDZapField(queueObj.ChangeID), auditQueueAttemptsZapField(queueObj.Attempts))
```

Especially since some of these field names aren't reference many time (e.g. `auditQueueAttemptsZapField`), it feels like it's adding a fair bit of bloat compared to just referencing infrequently used keys in plain text:

```go
logger.With(
    zap.String("queue_id", queueObj.ID),
    zap.Any("change_id", queueObj.ChangeID,
    zap.Int("queue_attempts", queueObj.Attempts)
)
```

Could also just be worth simplifying the approach by introducing a `package logfields` and exporting commonly used keys there:


```go
logger.With(
    zap.String(logfields.UserID, userID), // or even logfields.UserID(userID)
    zap.Any("change_id", queueObj.ChangeID,
    zap.Int("queue_attempts", queueObj.Attempts)
)
```

Reply:
I think a package could be a good idea, I think that I've introduced some keys in another package, and having a consistent shareable location would be more ideal.

I'll move them to their own package. I think I'd like to try the constructor functions there like `logfields.UserID(userID)`. I think spacing the with statement will help, as well as name spacing the log fields for context.

```go
logger.With(
      TranslatedAuditQueueIDZapField(queueObj.ID), 
      auditChangeIDZapField(queueObj.ChangeID), 
      auditQueueAttemptsZapField(queueObj.Attempts),
)
```